core/src/test/java/io/confluent/kafka/schemaregistry/rest/RestApiCompatibilityTest.java

```diff
@@ -642,4 +645,119 @@ public void testRegisterInvalidRuleSet() throws Exception {
+      assertEquals("Invalid schema", Errors.INVALID_SCHEMA_ERROR_CODE, rce.getErrorCode());
+    }
+  }
+
+  @Test
+  public void testSubjectAlias() throws Exception {
+    String subject = "testSubject";
+
+    // register a valid avro
+    String schemaString1 = AvroUtils.parseSchema("{\"type\":\"record\","
+        + "\"name\":\"myrecord\","
+        + "\"fields\":"
+        + "[{\"type\":\"string\",\"name\":\"f1\"}]}").canonicalString();
+    int expectedIdSchema1 = 1;
+    assertEquals("Registering should succeed",
+        expectedIdSchema1,
+        restApp.restClient.registerSchema(schemaString1, subject));
+
+    ConfigUpdateRequest config = new ConfigUpdateRequest();
+    config.setAlias("testSubject");
+    // set alias config
+    assertEquals("Setting alias config should succeed",
+        config,
+        restApp.restClient.updateConfig(config, "testAlias"));
+
+    Schema schema = restApp.restClient.getVersion("testAlias", 1);
+    assertEquals(schemaString1, schema.getSchema());
+  }
+
+  @Test
+  public void testSubjectAliasWithContext() throws Exception {
+    RestService restClient1 = new RestService(restApp.restConnect + "/contexts/.mycontext");
+    RestService restClient2 = new RestService(restApp.restConnect + "/contexts/.mycontext2");
+
+    // register a valid avro
+    String schemaString1 = AvroUtils.parseSchema("{\"type\":\"record\","
+        + "\"name\":\"myrecord\","
+        + "\"fields\":"
+        + "[{\"type\":\"string\",\"name\":\"f1\"}]}").canonicalString();
+    int expectedIdSchema1 = 1;
+    assertEquals("Registering should succeed",
+        expectedIdSchema1,
+        restClient1.registerSchema(schemaString1, "testSubject"));
+
+    ConfigUpdateRequest config = new ConfigUpdateRequest();
+    config.setAlias(":.mycontext:testSubject");
+    // set alias config
+    assertEquals("Setting alias config should succeed",
+        config,
+        restApp.restClient.updateConfig(config, ":.mycontext2:testAlias"));
+
+    Schema schema = restClient2.getVersion("testAlias", 1);
+    assertEquals(schemaString1, schema.getSchema());
+  }
+
+  @Test
+  public void testGlobalAliasNotUsed() throws Exception {
+    String subject = "testSubject";
+
+    // register a valid avro
+    String schemaString1 = AvroUtils.parseSchema("{\"type\":\"record\","
+        + "\"name\":\"myrecord\","
+        + "\"fields\":"
+        + "[{\"type\":\"string\",\"name\":\"f1\"}]}").canonicalString();
+    int expectedIdSchema1 = 1;
+    assertEquals("Registering should succeed",
+        expectedIdSchema1,
+        restApp.restClient.registerSchema(schemaString1, subject));
+
+    ConfigUpdateRequest config = new ConfigUpdateRequest();
+    config.setAlias("badSubject");
+    // set global alias config
+    assertEquals("Setting alias config should succeed",
+        config,
+        restApp.restClient.updateConfig(config, null));
+
+    Schema schema = restApp.restClient.getVersion("testSubject", 1);
```

Comment:
should this be `badSubject`?

Reply:
This test tests that aliases set at the global level (with no subject specified) don't do anything.  The test is called `testGlobalAliasNotUsed`.  In the future we may want to use these for something, but for now they are not used.